From 2d3835c1512359754052f2df12895240ee0112dc Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Thu, 12 Dec 2024 14:45:22 +0800
Subject: [PATCH 284/294] AOSCOS: spi: spi-loongson: allow building on
 MACH_LOONGSON32/64

LS2K/LS7A chipsets are also found on some MIPS-based Loongson hardware,
allow building for 32-bit (2K) and 64-bit (3A/B) hardware.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/spi/Kconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/spi/Kconfig b/drivers/spi/Kconfig
index 823797217404..75c23e6df830 100644
--- a/drivers/spi/Kconfig
+++ b/drivers/spi/Kconfig
@@ -542,12 +542,12 @@ config SPI_LM70_LLP
 
 config SPI_LOONGSON_CORE
 	tristate
-	depends on LOONGARCH || COMPILE_TEST
+	depends on LOONGARCH || MACH_LOONGSON32 || MACH_LOONGSON64 || COMPILE_TEST
 
 config SPI_LOONGSON_PCI
 	tristate "Loongson SPI Controller PCI Driver Support"
 	select SPI_LOONGSON_CORE
-	depends on PCI && (LOONGARCH || COMPILE_TEST)
+	depends on PCI && (LOONGARCH || MACH_LOONGSON32 || MACH_LOONGSON64 || COMPILE_TEST)
 	help
 	  This bus driver supports the Loongson SPI hardware controller in
 	  the Loongson platforms and supports to use PCI framework to
@@ -558,7 +558,7 @@ config SPI_LOONGSON_PCI
 config SPI_LOONGSON_PLATFORM
 	tristate "Loongson SPI Controller Platform Driver Support"
 	select SPI_LOONGSON_CORE
-	depends on OF && (LOONGARCH || COMPILE_TEST)
+	depends on OF && (LOONGARCH || MACH_LOONGSON32 || MACH_LOONGSON64 || COMPILE_TEST)
 	help
 	  This bus driver supports the Loongson SPI hardware controller in
 	  the Loongson platforms and supports to use DTS framework to
-- 
2.48.1

